### ISSUE TITLE
Pie Chart Label Position, require hybrid way of label positioning in labelLayout

### ISSUE THREAD

**Issue by taknirmal** [apache/echarts#19968]
### What problem does this feature solve?

So when using Pie chart, we are using Label positioning, label property in Pie chart has position and hideOverlap properties, 
Currently we have done customization on top of pie chart according to our requirement.
* We are using position inside for Pie chart which also has custom button created using label + click event,
* When we use labelLayout -> hideOverlap: true Chart looks like below:

![image](https://github.com/apache/echarts/assets/123442568/9370bb3c-11b2-4067-9352-fa13a999cf75)

* When we uselabelLayout -> hideOverlap: false Chart looks like below, which is mess.

![image](https://github.com/apache/echarts/assets/123442568/44b15dc4-8fe1-47d6-b0eb-c35d6cbb342c)

so we need a way to push the label which are hidden due to overlapping to be displayed out side and not all.

![image](https://github.com/apache/echarts/assets/123442568/6f0528fe-1884-4554-8cdb-cb5145ee1a27)

so basically we need mixture of first scenario and 3rd scenario, instead of hiding labels due to overlap, can we push them outside.

### What does the proposed API look like?

labelLayout in Pie chart, can have another property which allows labels to be pushed outside when overlapping.
